test/unit/parser_spec.js

```diff
@@ -14,110 +14,182 @@
  */
 /* eslint no-var: error */
 
-import { Lexer, Linearization } from '../../src/core/parser';
+import { Lexer, Linearization, Parser } from '../../src/core/parser';
 import { FormatError } from '../../src/shared/util';
 import { Name } from '../../src/core/primitives';
 import { StringStream } from '../../src/core/stream';
 
 describe('parser', function() {
-  describe('Lexer', function() {
-    it('should stop parsing numbers at the end of stream', function() {
-      const input = new StringStream('11.234');
-      const lexer = new Lexer(input);
-      expect(lexer.getNumber()).toEqual(11.234);
-    });
+  describe('Parser', function() {
+    describe('inlineStreamSkipEI', function() {
+      it('should skip over the EI marker if it is found', function() {
+        const string = 'BI /foo /bar EI<<';
```

Comment:
While it may not matter for these particular tests, please note that these commands do *not* represent a valid content stream :-P

Unless you want to test for a *specific* mode of corruption, as found in real-world PDFs, it's always best if the unit-test is actually correct in my experience; please find an actual example below (from 10388_reduced.pdf). 
```
BI 
/W 16
/H 17
/BPC 1
/IM true 
ID ÿÿÿÿþþþþþþÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿÿ
EI
```

Reply:
That's a fair point. In this case it didn't really matter, but I agree that it's probably better to use real data instead of fake data here.

Reply:
*Apparently I wrote that code, so here a bit of unsolicited advice :-)*

As the comment mentions this method is only used for inline image streams with an EOD (end-of-data) marker, i.e. `DCTDecode` (== JPEG), `ASCII85Decode`, and `ASCIIHexDecode`; see https://github.com/mozilla/pdf.js/blob/d587abbceb97008ed744261c219c90d3ce981f2b/src/core/parser.js#L390-L393
When this method is called, the EOD marker in question has been successfully found and we simply want to advance the stream position just after the `EI` command (ignoring any white-space in between).

The example above wasn't that great in hindsight, since in practice it wouldn't apply here (no /Filter entry, and thus no EOD marker present).
The following is unlikely to be actually valid image data, so please keep that in mind, but let's consider the following example instead:
```
q
1 0 0 1 0 0 cm 
BI 
/W 10
/H 10
/F /A85 
ID abc123~>
EI
Q
```
When the main part of `findASCII85DecodeInlineStreamEnd` has run, the stream position is now just after `~>`. With the help of `inlineStreamSkipEI` the stream position will be advanced to just after `EI`, which is what you need here.

Reply:
Fixed in the new commit series. The unit test cases are now more like what we encounter in real streams. I couldn't find a nice way to put binary image data in there and I think the mock value `abc123` is fine since our only purpose is to show that we can skip over the `EI` marker, so the actual binary data seems irrelevant here.